### PR TITLE
Add cypress v5 definitions

### PR DESCRIPTION
**Pull request by Brianzchen**
- Links to documentation: https://docs.cypress.io/guides/overview/why-cypress.html
- Link to GitHub or NPM: https://github.com/cypress-io/cypress
- Type of contribution: new definition

Other notes: Thanks to @lukeapage for #3028 which was a great starting base

### CODE REVIEW

**Comment by Brianzchen** [comment]
I still need to add more tests. Just opening PR now to get early feedback

File: definitions/npm/cypress_v5.x.x/flow_v0.25.x-/cypress_v5.x.x.js [hunk 1]
```diff
@@ -0,0 +1,803 @@
...
+declare type Cypress$Env = { [key: string]: string, ... };
+
+declare type Cypress$Window = any;
+
+declare interface Cypress$Core {
+  /**
+   *
+   * @see https://docs.cypress.io/api/cypress-api/custom-commands.html
+   */
+  addChildCommand(name: string, fn: Function): void,
```
Comment (pascalduez) [inline definitions/npm/cypress_v5.x.x/flow_v0.25.x-/cypress_v5.x.x.js:65]: `Function` has been aliased to `any` from Flow 0.88.0.

**suggestion:** replace with a real function type?

File: definitions/npm/cypress_v5.x.x/flow_v0.25.x-/cypress_v5.x.x.js [hunk 2]
```diff
@@ -0,0 +1,803 @@
...
+  config(key: string, value: any): void,
+  config(object: { ... }): void,
+
+  /**
+   * @see https://docs.cypress.io/api/cypress-api/env.html
+   */
+  env(): Cypress$Env,
+  env(name: string): string,
+  env(name: string, value: string): void,
+  env(object: Cypress$Env, ...rest: Array<void>): void,
```
Comment (pascalduez) [inline definitions/npm/cypress_v5.x.x/flow_v0.25.x-/cypress_v5.x.x.js:84]: I guess typing rest params is more like `...rest: Array<any>`.

File: definitions/npm/cypress_v5.x.x/flow_v0.25.x-/cypress_v5.x.x.js [hunk 3]
```diff
@@ -0,0 +1,803 @@
...
+    isElement: (element: any) => boolean,
+    isFocusable: (element: any) => boolean,
+    isFocused: (element: any) => boolean,
+    isHidden: (element: any) => boolean,
+    isJquery: (element: any) => boolean,
+    isScrollable: (element: any) => boolean,
+    isVisible: (element: any) => boolean,
+    isWindow: (element: any) => boolean,
+    unwrap: (element: any) => Array<any>,
+    wrap: (element: any) => any,
```
Comment (pascalduez) [inline definitions/npm/cypress_v5.x.x/flow_v0.25.x-/cypress_v5.x.x.js:163]: **suggestion:** can't all those `any` be replaced with something like `Element` or `HTMLElement`?

File: definitions/npm/cypress_v5.x.x/flow_v0.25.x-/cypress_v5.x.x.js [hunk 4]
```diff
@@ -0,0 +1,803 @@
...
+  body?: Cypress$RequestBody,
+  failOnStatusCode?: boolean,
+  followRedirect?: boolean,
+  form?: boolean,
+  gzip?: boolean,
+  headers?: Object,
+  method?: Cypress$HttpMethod,
+  qs?: string,
+  url: string
+|} & Cypress$Loggable & Cypress$Timeoutable
```
Comment (pascalduez) [inline definitions/npm/cypress_v5.x.x/flow_v0.25.x-/cypress_v5.x.x.js:693]: **suggestion:** prefer spread to intersection for exact objects.

**Review by pascalduez — CHANGES_REQUESTED** [review]
Thanks for this!

A bunch of `Function` that needs to be replaced by: ideally a real type, otherwise `any`.

File: definitions/npm/cypress_v5.x.x/flow_v0.25.x-/cypress_v5.x.x.js (hunk 3 above)
Comment (Brianzchen) [reply definitions/npm/cypress_v5.x.x/flow_v0.25.x-/cypress_v5.x.x.js:163]: Based on how this usage is documented,
```
cy.get('img').then(($el) => {
  Cypress.dom.isVisible($el) // true
})
```

I tried the following in my code
```
cy.get(selectors.loginLink).then((ele) => {
  console.log(ele);
});
```
Which gave me
Screen Shot 2020-09-25 at 8 32 56 am

From what I can tell cypress has their own wrapped objects for all these things. Even in their docs when it says it should return a dom element, they actually mean it's an instance of `Cypress$Global` which annoyed me so much when typing it

File: definitions/npm/cypress_v5.x.x/flow_v0.25.x-/cypress_v5.x.x.js (hunk 1 above)
Comment (Brianzchen) [reply definitions/npm/cypress_v5.x.x/flow_v0.25.x-/cypress_v5.x.x.js:65]: Wow! I didn't know that, this is so misleading. I found the docs for it so I'll make the changes to use `(...args: Array<any>) => any` instead

https://flow.org/en/docs/types/functions/#toc-function-type

File: definitions/npm/cypress_v5.x.x/flow_v0.25.x-/cypress_v5.x.x.js (hunk 4 above)
Comment (Brianzchen) [reply definitions/npm/cypress_v5.x.x/flow_v0.25.x-/cypress_v5.x.x.js:693]: Do you mean to say, change it to this instead?

For some reason when I do this, method overloading doesn't work anymore for the functions that use this type as an option, any idea how I could fix this?
```
declare type Cypress$RequestOptions = {|
  ...Cypress$Loggable,
  ...Cypress$Timeoutable,
  auth?: Object,
  body?: Cypress$RequestBody,
  failOnStatusCode?: boolean,
  followRedirect?: boolean,
  form?: boolean,
  gzip?: boolean,
  headers?: Object,
  method?: Cypress$HttpMethod,
  qs?: string,
  url: string
|}
```

**Review by pascalduez — APPROVED** [review]

**Comment by pascalduez** [comment]
Okay let's move forward and improve in later PRs if needed.

Thanks!

**Pull request merged**